### ISSUE TITLE
Un-authed -> am-i-used -> auth -> am-i-used fails

### ISSUE THREAD

**Issue by Qix-** [yamadapc/am-i-used#2]
Couldn't think of much of a better title.

I reformatted lately and forgot that I hadn't logged into NPM since I did. I installed `am-i-used` and tried to use it, and it greeted me with `/usr/local/bin/am-i-used` as the user.

I `npm adduser`'d and tried again; still `/usr/local/bin/am-i-used`. Reinstalling doesn't help.

**Comment by yamadapc** [comment]
Mm... What does `npm whoami` give you?

**Comment by Qix-** [comment]
``` shell
$ npm whoami
qix
```

**Comment by Qix-** [comment]
Something else weird is going on here.

```
$ node
> function getCurrentNpmUser(cb) {
...   child.exec('npm whoami', [], function(err, stdout/*, stderr*/) {
.....     if(err) cb(err);
.....     else cb(null, stdout.toString().slice(0, -1));
.....   });
... }
undefined
> var child = require('child_process');
undefined
> getCurrentNpmUser(console.log.bind(console));
undefined
> null 'qix'
```

**Comment by Qix-** [comment]
Weirdly enough I think it's this line of code.

``` javascript
if(args.length) {
        return eachSeries(args, reportUse, exit);
 }
```

`args.length` will usually always be at least 1, since Node passes in the script name at the very beginning of the argument list. If it's invoked using `node index.js` then it'll have two; `node` and then the script path. You're using a shebang, so only the script has to be invoked for it to work because you're relying on the shell to understand the shebang. Nothing wrong with that, other than it doesn't work on windows :)

`if (args.length)` will always evaluate to true, hence the output `Hello, /usr/local/bin/am-i-used`.

**Comment by Qix-** [comment]
Oops, I see what you're doing. You _are_ slicing argv; however, logging argv produces:

```
[ '/usr/local/bin/node', '/usr/local/bin/am-i-used' ]
```

There's the problem :dancer: `/[\\\/]node$/i.test(process.argv[0])` perhaps.

**Comment by yamadapc** [comment]
Fixed in `0.0.7`. I think this changed recently on node.js, not the first time I fix this problem this week. :P

**Comment by Qix-** [comment]
That wouldn't surprise me. Thanks! :dancer:
